client/landing/stepper/declarative-flow/hosted-site-migration-v2/index.tsx

```diff
@@ -11,14 +18,75 @@ export default {
 	},
 	isSignupFlow: false,
 	useSteps() {
-		return stepsWithRequiredLogin( [ STEPS.IMPORT_OR_MIGRATE_WITH_UPGRADE ] );
+		return stepsWithRequiredLogin( [
+			PLATFORM_IDENTIFICATION,
+			SITE_CREATION_STEP,
+			PROCESSING,
+			SITE_MIGRATION_UPGRADE_PLAN,
+		] );
 	},
 
-	//TEMP: This is a temporary implementation to make the flow work
-	// eslint-disable-next-line @typescript-eslint/no-unused-vars
 	useStepNavigation( currentStep, navigate ) {
+		const [ query ] = useSearchParams();
+
 		return {
-			submit: () => alert( 'Go' ),
+			submit: ( props?: ProvidedDependencies ) => {
+				const data = {
+					platform: props?.platform || query.get( 'platform' ),
+					siteId: props?.siteId || query.get( 'siteId' ),
+					siteSlug: props?.siteSlug || query.get( 'siteSlug' ),
+					next: props?.next || query.get( 'next' ),
+					url: props?.url,
+				} as Record< string, Primitive >;
+
+				if ( currentStep === PLATFORM_IDENTIFICATION.slug ) {
+					return navigate(
+						addQueryArgs(
+							{
+								platform: data.platform,
+								next: data.url,
+							},
+							STEPS.SITE_CREATION_STEP.slug
+						)
+					);
+				}
+
+				if ( currentStep === SITE_CREATION_STEP.slug ) {
+					return navigate(
+						addQueryArgs(
+							{
+								platform: data.platform,
+								next: data.next,
+							},
+							STEPS.PROCESSING.slug
+						)
+					);
+				}
+
+				if ( currentStep === PROCESSING.slug ) {
+					if ( data?.next ) {
+						return window.location.assign(
+							addQueryArgs(
+								{
+									siteId: data?.siteId,
+									siteSlug: data?.siteSlug,
+								},
+								`/setup/${ SITE_SETUP_FLOW }/${ data.next.toString() }`
```

Comment:
When redirecting to this flow, I'm receiving this error:

```
Tracks: Event `calypso_stepper_flow_start` will be rejected because property name `siteId` does not match /^[a-z_][a-z0-9_]*$/. Please use a compliant property name.
```

Is it the issue you mentioned that is happening when the flow starts that someone is fixing? Or should we worry about this in this project?

Reply:
PR fixing it #93364